broker/src/main/java/org/apache/rocketmq/broker/processor/ClientManageProcessor.java

```diff
@@ -138,6 +144,63 @@ public RemotingCommand heartBeat(ChannelHandlerContext ctx, RemotingCommand requ
         }
         response.setCode(ResponseCode.SUCCESS);
         response.setRemark(null);
+        response.addExtField(MixAll.IS_SUPPORT_HEART_BEAT_V2, Boolean.TRUE.toString());
+        response.addExtField(MixAll.IS_SUB_CHANGE, Boolean.TRUE.toString());
+        return response;
+    }
+
+    private RemotingCommand heartBeatV2(ChannelHandlerContext ctx, HeartbeatData heartbeatData, ClientChannelInfo clientChannelInfo, RemotingCommand response) {
+        boolean isSubChange = false;
+        for (ConsumerData consumerData : heartbeatData.getConsumerDataSet()) {
+            //Reject the PullConsumer
+            if (brokerController.getBrokerConfig().isRejectPullConsumerEnable()) {
+                if (ConsumeType.CONSUME_ACTIVELY == consumerData.getConsumeType()) {
+                    continue;
+                }
+            }
+            if (null != consumerGroupHeartbeatTable.get(consumerData.getGroupName()) && consumerGroupHeartbeatTable.get(consumerData.getGroupName()) != heartbeatData.getHeartbeatFingerprint()) {
```

Comment:
Suggest line breaks for better codestyle

Reply:
Thank you for your review,  one line is also a kind of beauty～

Reply:
Could use StringUtils.equals()